transform/transformers/IndexFile.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+import dateutil.parser
+
+from io import BytesIO
+from transform import settings
+from transform.views.image_filters import get_env, format_date
+
+
+class IndexFile:
+    """Class for creating in memory index_file file using StringIO."""
+
+    def __init__(self, logger, response_data, image_count, image_names,
+                 current_time=None, sequence_no=1000):
+
+        if current_time is None:
+            current_time = datetime.datetime.utcnow()
+
+        self.in_memory_index = BytesIO()
+        self.logger = logger
+        self._response = response_data
+        self._image_count = image_count
+        self._creation_time = {
+            'short': format_date(current_time, 'short'),
+            'long': format_date(current_time)
+        }
+        self.index_name = self._get_index_name(self._response, sequence_no)
+        self._current_time = current_time  # used to test if current_time gets set to a default value in init definition
+        self._build_index(image_names)
+
+    def rewind(self):
+        """Rewinds the read-write position of the in-memory in_memory_index to the start."""
```

Comment:
Does the rewind method need to be "public", it seems like it is never called externally?

Reply:
Its currently not used externally but is a function that external client code code use hence public . But I can see it both ways